### PR TITLE
Added forced Host update button to webfrontend

### LINKED ISSUE #996: Hosts: Check connection button
The webfrontend should have a button which checks connection to the host.
This is useful to speed up maintenance in case of connection problems.

### PR DESCRIPTION
**Pull request by swuest**
Works and was tested in Firefox and Chrome. It also should updated host_list and host_info in the cache, but I have no clue why this won't work. But it is not a bug or a real problem for this feature and can be solved in a different issue.

Closes #996

### CODE REVIEW

File: backend/tomato/host/__init__.py [hunk 1]
```diff
@@ -175,21 +176,40 @@ def incrementErrors(self):
...
 			self.hostNetworks = self.getProxy().host_networks()
 		except:
 			self.hostNetworks = []
-		caps = self._convertCapabilities(self.getProxy().host_capabilities())
-		self.elementTypes = caps["elements"]
-		self.connectionTypes = caps["connections"]
+	
+		oldElementTypes = self.elementTypes
+		oldconnectionTypes = self.connectionTypes
+		caps = []
```
Comment (dswd) [inline backend/tomato/host/__init__.py]: caps is expected to be a dict

File: backend/tomato/host/__init__.py [hunk 2]
```diff
@@ -100,6 +100,7 @@ class Host(DumpSource, Entity, BaseDocument):
 	dumpLastFetch = FloatField(db_field='dump_last_fetch')
 	templates = ListField(ReferenceField(Template, reverse_delete_rule=PULL))
 	hostNetworks = ListField(db_field='host_networks')
+	synchronized = True
```
Comment (dswd) [inline backend/tomato/host/__init__.py]: This does not work. This variable is not stored in the database and will be lost when the object is fetched by a different operation.

File: backend/tomato/api/host.py [hunk 3]
```diff
@@ -154,11 +154,19 @@ def host_create(name, site, attrs=None):
 	return h.info()
 
 @checkauth
-def host_info(name):
+def host_info(name,force_update=False):
 	"""
 	undocumented
 	"""
 	h = _getHost(name)
+	if(force_update):
```
Comment (dswd) [inline backend/tomato/api/host.py]: Only host_managers should be able to do this

**Comment by dswd** [comment]
What are the reasons behind the semantic change of update (problem entry instead of exception)?

File: backend/tomato/host/__init__.py [hunk 4]
```diff
@@ -175,27 +177,34 @@ def incrementErrors(self):
...
-		caps = self._convertCapabilities(self.getProxy().host_capabilities())
-		self.elementTypes = caps["elements"]
-		self.connectionTypes = caps["connections"]
-		self.componentErrors = max(0, self.componentErrors / 2)
-		if not self.problems():
-			self.availability += 1.0 - config.HOST_AVAILABILITY_FACTOR
-		self.save()
-		logging.logMessage("info", category="host", name=self.name, info=self.hostInfo)
-		logging.logMessage("capabilities", category="host", name=self.name, capabilities=caps)
+			self.synchronized = False
```
Comment (t-gerhard) [inline backend/tomato/host/__init__.py]: You should raise the exception after setting the flag to keep the original semantics.

**Comment by dswd** [comment]
Superseeded by new code

**Pull request closed without merging**